### PR TITLE
feat: Introducing AWS Lambda based WES Adapter for running the workflows

### PR DESCRIPTION
**Pull request by AndreyDovydenko**
**Description of Changes**

Replacing container running service for WES adapter with a Lambda based one.

**Description of how you validated changes**

I have run the workflows from the example folders for nextflow, cromwell and miniwdl engines, all of which succeeded.

**Checklist**

- [x] If this change would make any existing documentation invalid, I have included those updates within this PR
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have linted my code before raising the PR
- [x] Title of this Pull Request follows Conventional Commits standard: https://www.conventionalcommits.org/en/v1.0.0/

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by tneely** [comment]
You have an empty `packages/cdk/lib/wes_adapter/rest_api/controllers/security_controller_.py` file committed. Can it be deleted?

File: packages/cdk/lib/wes_adapter/amazon_genomics/wes/adapters/CromwellWESAdapter.py [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+        if not contents:
+            raise RuntimeError("empty workflow.zip")
+
+        if len(contents) == 1:
+            # single file workflow
+            files["workflowSource"] = open(path.join(wd, contents[0]), "rb")
+
+        else:
+            # multifile workflow
+            if "MANIFEST.json" in contents:
```
Comment (tneely) [inline packages/cdk/lib/wes_adapter/amazon_genomics/wes/adapters/CromwellWESAdapter.py:567]: I'm not seeing support for multi file workflows in the Batch/Nextflow/MiniWdl adapters. Does it just work out of the box or can those engines not support a MANIFEST.json?

**Review by tneely — CHANGES_REQUESTED** [review]

**Comment by AndreyDovydenko** [comment]
> You have an empty `packages/cdk/lib/wes_adapter/rest_api/controllers/security_controller_.py` file committed. Can it be deleted?

I think we could, the files under `rest_api` are mostly auto-generated and moved over from our previous code base for the WES. I'm okay with taking it down.

File: packages/cdk/lib/stacks/nested/miniwdl-engine-stack.ts [hunk 2]
```diff
@@ -79,15 +80,17 @@ export class MiniWdlEngineStack extends NestedEngineStack {
...
+      role: adapterRole,
+      jobQueueArn: this.batchHead.jobQueue.jobQueueArn,
+      jobDefinitionArn: this.miniwdlEngine.headJobDefinition.jobDefinitionArn,
+    });
+    this.adapterLogGroup = lambda.logGroup;
 
     this.apiProxy = new ApiProxy(this, {
-      apiName: `${params.projectName}${params.userId}${params.contextName}MiniWdlApiProxy`,
-      loadBalancer: adapter.loadBalancer,
+      apiName: `${params.projectName}${params.userId}${params.contextName}NextflowApiProxy`,
```
[the remark this answers is not in the document]
Comment (AndreyDovydenko) [reply packages/cdk/lib/stacks/nested/miniwdl-engine-stack.ts:92]: Good catch, thanks!

File: packages/cdk/lib/wes_adapter/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Amazon Genomics Workflow Execution Service (WES) Adapter
+
+A basic GA4GH compliant Workflow Execution Service that enables use of adaptors for integrating with workflow execution engines.
+The code in this package is capturing the Lambda Function source code that's deployed as WES adapter endpoint for purposes of AGC
+
+
+# Open source usage
+This package contains the source code  that will be built and packaged before uploading into your AWS account as lambda code; this upload will contain the open source project dependencies shown in the requirement.txt file
```
[the remark this answers is not in the document]
Comment (AndreyDovydenko) [reply packages/cdk/lib/wes_adapter/README.md:8]: :)

File: packages/cdk/lib/wes_adapter/amazon_genomics/wes/adapters/CromwellWESAdapter.py (hunk 1 above)
Comment (AndreyDovydenko) [reply packages/cdk/lib/wes_adapter/amazon_genomics/wes/adapters/CromwellWESAdapter.py:567]: Right, so the difference here is that Nextflow/Miniwdl handle that within the head job container. Cromwell adapter is quite different compared to the other two we have.

File: packages/cdk/lib/roles/nextflow-adapter-role.ts [hunk 4]
```diff
@@ -11,10 +12,22 @@ export interface NextflowAdapterRoleProps extends NextflowSubmitJobBatchPolicyPr
...
+      },
+      scope as Stack
+    );
+    const nextflowJobArn = Arn.format(
+      {
+        resource: "job/*",
+        service: "batch",
+      },
+      scope as Stack
+    );
```
[the remark this answers is not in the document]
Comment (AndreyDovydenko) [reply packages/cdk/lib/roles/nextflow-adapter-role.ts:28]: Looks like not all of the code got merged well when i moved it in here, thanks!

**Review by tneely — APPROVED** [review]

**Review by IllyaYalovyy — APPROVED** [review]

**Pull request merged**